### DISCUSSION TITLE
Is there a difference between SimpleSelectStatement.get() and JoinedSelectStatement.get() ? Which is more efficient and performant?

### DISCUSSION THREAD

**Discussion by ddxl123** [simolus3/drift#2055] [Q&A] [answered]
Is there a difference between `SimpleSelectStatement.get()` and `JoinedSelectStatement.get()` ? 
Which is more efficient and performant?
```
final query = select( tableA )
      ..join( [  innerJoin( tableB, tableB.aId.equalsExp(tableA .id) ) ],)
      ..where( (tbl) => tbl.id.equals( aId ) );
final List<TableA> finalResult = await query.get();
```
```
final query = select( tableA ).join( [ innerJoin( tableB, tableB.aId.equalsExp( tableA.id ) ) ] )
      ..where( tableB.id.equals( bId ) );
final List<TypedResult> result = await query.get();
final List<TableA> finalResult = result.map((e) => e.readTable(fragments)).toList();
```
It feels that the second one needs to be iterated again, so the first one should be more efficient?

**Comment by simolus3** [comment, marked as answer]
In the first query, you're not actually running a join. You need to use the return value of `.join` to further build the query for it to be a join. So the cascaded `join` call in the first query has no effect at all.

If your intention is to only use the second table (`tableB`) for the `where` predicate and you don't care about any of its columns, you can set `useColumns: false`. You still need the mapping, but I think that won't affect the overall performance much.
